core/src/test/scala/kafka/raft/KafkaMetadataLogTest.scala

```diff
@@ -108,12 +116,57 @@ final class KafkaMetadataLogTest {
       classOf[RuntimeException],
       () => {
         log.appendAsFollower(
-          MemoryRecords.withRecords(initialOffset, Compression.NONE, currentEpoch, recordFoo)
+          MemoryRecords.withRecords(initialOffset, Compression.NONE, currentEpoch, recordFoo),
+          currentEpoch
         )
       }
     )
   }
 
+  @Test
+  def testEmptyAppendNotAllowed(): Unit = {
+    val log = buildMetadataLog(tempDir, mockTime)
+
+    assertThrows(classOf[IllegalArgumentException], () => log.appendAsFollower(MemoryRecords.EMPTY, 1));
+    assertThrows(classOf[IllegalArgumentException], () => log.appendAsLeader(MemoryRecords.EMPTY, 1));
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(classOf[InvalidMemoryRecordsProvider])
```

Comment:
maybe I'm missing something - should we also have a test here where we call `log.appendAsFollower(records, epoch)` where `epoch` is less than one of the epochs in the records batch? it could be a malformed batch and we could check that the test does not throw CorruptRecordException

Reply:
Yes. You are correct, we need to test the case when the leader epoch is invalid. I added tests for that case to `MockLogTest`, `KafkaMetadataLogTest` and `UnifiedLogTest`